text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,236 @@
+errors to be tied back to original module version and source code.
+
+Defining the specification for Environment Variables as carriers will have a
+wide impact to the industry in enabling better observability to systems outside
+of the normal HTTP microservice architecture.
+
+[w3c-parent]: https://www.w3.org/TR/trace-context/#traceparent-header
+[w3c-state]: https://www.w3.org/TR/trace-context/#tracestate-header
+[w3c-bag]: https://www.w3.org/TR/baggage/#header-name
+[bag-api]: https://opentelemetry.io/docs/specs/otel/baggage/api/
+
+## Internal details
+
+Defining the specification for Environment Variables as carriers for context
+will enable SDK's and other tools to implement getters and settings of context
+in a standard, observable way. Therefore, current OpenTelemetry language
+maintainers will need to develop language specific implementations that adhere
+to the specification. Of course the specification itself will have to be
+defined, and a decision made on whether or not to adhere to the W3C
+specification or not will need to be made.
+
+Due to the current programming conventions, prior art, and information below,
+it is recommended to leverage upper-cased environment variables for the
+carrier. `TextMapPropagator` will be the means of propagation for environment
+variables.
+
+### UNIX
+
+UNIX system utilities use upper-case for environment variables and lower-case
+are reserved for applications. Using upper-case will prevent conflicts with
+internal application variables.
+
+Environment variable names used by the utilities in the XCU specification
+consist solely of upper-case letters, digits and the "_" (underscore) from the
+characters defined in Portable Character Set. Other characters may be permitted
+by an implementation; applications must tolerate the presence of such names.
+Upper- and lower-case letters retain their unique identities and are not folded
+together. The name space of environment variable names containing lower-case
+letters is reserved for applications. Applications can define any environment
+variables with names from this name space without modifying the behaviour of
+the standard utilities.
+
+Source: [The Open Group, The Single UNIX® Specification, Version 2, Environment Variables](https://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html)
+
+### Windows
+
+Windows is case-insensitive with environment variables. Despite this, the
+recommendation to be upper-cased across OS.
+
+Some languages already do this. This [CPython issue][cpython] discusses how
+Python automatically upper-cases environment variables. The issue was merged and
+this [documentation][cpython-doc] was added to clarify the behavior.
+
+[cpython]: https://github.com/python/cpython/issues/101754
+[cpython-doc]: https://docs.python.org/3/library/os.html#os.environ
+
+### Allowed characters
+
+The characters allowed in keys by the `TextMapPropagator` are also allowed as
+environment variable keys on UNIX and Windows.
```

Comment:
I am not following this. Are you saying that say `&` is allowed in the name of the environment variable?

Reply:
That's the way it reads. This was one of the original statements. I've removed It. From what I'm reading, the only values allowed in the keys are letters, numbers and hyper/underscore. Updating this to reflect.